### PR TITLE
devguide: discuss backports and other updates - v2

### PR DESCRIPTION
**Pull request by jufajardini**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6568

Describe changes:
- add notes for exceptional cases
- change format for `Needs backport` label indication
- reword references to stable releases
- include link to EOL policies
- fix reference link to dev guide Readme

Result can be seen at: https://suri-rtd-test.readthedocs.io/en/contribution-docs-updates-v2/devguide/contributing/backports-guide.html

### CODE REVIEW

File: doc/userguide/devguide/contributing/backports-guide.rst [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+Usually, when the team creates a ticket, we'll add the *Needs backport* related
+labels, so necessary backporting tickets will be automatically created. If you
+are working on a ticket that doesn't have such labels, nor backporting tasks
+associated, it probably doesn't need backporting. If you understand that the
+issue should be backported, please let us know in the ticket or related PR. But
+sometimes we'll miss those.
+
+The general principle used to determine what will be backported is:
+    * evasion/security fixes
```
Comment (victorjulien) [inline doc/userguide/devguide/contributing/backports-guide.rst:36]: I would just call these "security fixes" and point to our `SECURITY.md` for our definitions of security tickets and their severity levels

File: doc/userguide/devguide/contributing/backports-guide.rst [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    * behavioral change: how much will the behavior of the system be changed by the
+      backport. for example, a small change to decode additional encapsulation
+      protocols may result in more traffic being presented to Suricata.
+    * default settings: if the issue alters behavior, can it be made optional, and
+      at what cost?
+
+Creating backport tickets -- new issues
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Redmine: for security/evasion and bug fixes, when creating a new redmine issue,
```
Comment (victorjulien) [inline doc/userguide/devguide/contributing/backports-guide.rst:64]: we have a security ticket type, so lets just refer to that and not add evasion here (and in other places)

File: doc/userguide/devguide/contributing/backports-guide.rst [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+current Suricata release may not apply to older releases.
+
+Git Backport Workflow
+---------------------
+
+    * *Identify the commit(s) needed* for the backport. Start with the PR that merged
+      the commits into master and select only the commits from the issue being
+      backported.
+    * *Bring each commit into the new branch,* one at a time -- starting with the oldest
+      commit. Use ``git cherry-pick -x commit-hash``, where ``commit-hash`` is
```
Comment (victorjulien) [inline doc/userguide/devguide/contributing/backports-guide.rst:93]: this should mention that the hashes should already be in `master` or `main-7.0.x` and that we have a CI check that checks for the validity of the cherry-pick line

File: doc/userguide/devguide/contributing/code-submission-process.rst [hunk 4]
```diff
@@ -9,16 +9,18 @@ Commits
...
-   for each of this, e.g, a commit to rename files, then a commit to change the code.
-#. Documentation updates should be in their own commit (not mixed with code commits)
-#. Commit messages need to be properly formatted:
-    * Meaningful and short (50 chars max) subject line followed by an empty line
-    * Naming convention: prefix message with sub-system ("rule parsing: fixing foobar"). If
-      you're not sure what to use, look at past commits to the file(s) in your PR.
-    * Description, wrapped at ~72 characters
+#. Commits should not, at the same time, change, rename and/or move code. Use separate commits
+     for each of this, e.g, a commit to rename files, then a commit to change the code.
+#. If your code changes or adds new behavior, update the documentation in the same commit.
```
Comment (victorjulien) [inline doc/userguide/devguide/contributing/code-submission-process.rst:14]: I guess I don't feel very strongly about this anymore. For backporting it is generally easier if commits are small and not touching too many things at once.

File: doc/userguide/devguide/contributing/code-submission-process.rst [hunk 5]
```diff
@@ -9,16 +9,18 @@ Commits
...
+     Otherwise, documentation updates should be in their own commit (not mixed with code commits).
+#. Commit messages need to be properly formatted (check the example further
```
Comment (victorjulien) [inline doc/userguide/devguide/contributing/code-submission-process.rst:16]: we have a separate doc for this?

File: doc/userguide/devguide/contributing/github-pr-workflow.rst [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+.. note:: It is expected that the author will create a new PR with a new version
+   of the patch as described in :ref:`Pull Requests Criteria <pull-requests-criteria>`.
+   A PR may be closed as stale if it has not been updated in two months after
+   changes were requested.
+
+A PR may be labeled *decision-required* if the reviewer thinks the team needs
+more time to analyze the best approach to a proposed solution or discussion
+raised by the PR.
+
+Once in approved state, the PRs are in the responsibility of the merger, along
```
Comment (victorjulien) [inline doc/userguide/devguide/contributing/github-pr-workflow.rst:40]: "merger" -> maintainer?

**Review by victorjulien — CHANGES_REQUESTED** [review]
some feedbacks inline

File: doc/userguide/devguide/contributing/code-submission-process.rst (hunk 4 above)
Comment (jufajardini) [reply doc/userguide/devguide/contributing/code-submission-process.rst:14]: This part isn't for the backports guide, should I separate this part of the work in a different PR?

File: doc/userguide/devguide/contributing/code-submission-process.rst (hunk 5 above)
Comment (jufajardini) [reply doc/userguide/devguide/contributing/code-submission-process.rst:16]: From memory, and my checks, so far, this has been the place where we mentioned this stuff. Want me to break it into another document?

File: doc/userguide/devguide/contributing/code-submission-process.rst (hunk 4 above)
Comment (jufajardini) [reply doc/userguide/devguide/contributing/code-submission-process.rst:14]: Another thought: although I see your point, I feel it is easier to miss a documentation portion of a code-change being backported if it is in a separate commit...

Comment (victorjulien) [reply doc/userguide/devguide/contributing/code-submission-process.rst:14]: I find smaller changes per commit easier to backports. Reduces chances of merge conflicts and makes it easier to spot merge/rebase issues.

File: doc/userguide/devguide/contributing/code-submission-process.rst (hunk 5 above)
Comment (victorjulien) [reply doc/userguide/devguide/contributing/code-submission-process.rst:16]: no I think this is the doc I remembered, just not realized we were talking about it now ;)

**Comment by jufajardini** [comment]
Replaced by: https://github.com/OISF/suricata/pull/10056

**Pull request closed without merging**